pkgs/applications/networking/instant-messengers/element/element-desktop.nix

```diff
@@ -80,7 +86,7 @@ mkYarnPackage rec {
     # executable wrapper
     makeWrapper '${electron_exec}' "$out/bin/${executableName}" \
       --add-flags "$out/share/element/electron${lib.optionalString useWayland " --enable-features=UseOzonePlatform --ozone-platform=wayland"}"
-  '';
+  '' + lib.optionalString stdenv.isDarwin darwinApplicationCmds + darwinIconCmds;
```

Comment:
As been already pointed out, this should become a setup hook. It should be easy to implement, and will make things more consistent with how we normally do such things.